src/summarize.jl

```diff
@@ -158,10 +160,16 @@ function summarize(
     # Obtain the additional named tuple.
     additional_nt = additional_df === nothing ? NamedTuple() : additional_df.nt
 
+    # TODO: Handle multiple weighted chains.
     if append_chains
         # Evaluate the functions.
         data = to_matrix(chn)
-        fvals = [[f(data[:, i]) for i in axes(data, 2)] for f in funs]
+
+        if chains.weights isa StatsBase.UnitWeights
+            fvals = [[f(data[:, i]) for i in axes(data, 2)] for f in funs]
+        else 
+            fvals = [[f(data[:, i], chains.weights) for i in axes(data, 2)] for f in funs]
+        end
```

Comment:
It would be easier here to handle this by dispatching on `weights`.
```suggestion
        fvals = [[f(data[:, i], chains.weights) for i in axes(data, 2)] for f in funs]
```

Reply:
As in, the user should pass functions that ignore the passed weights? That would definitely be easier but causes problems for functions that don't have a method taking `weights`.

Reply:
All functions that we use in MCMCChains have to support weights anyway before this PR is merged. And I think it's reasonable to demand the same from any user-provided functions since one can just ignore the weights if one only uses unweighted samples (not sure if anyone actually uses this function and if it should be exported but that's a separate issue).

Reply:
I mean, maybe it shouldn't be, but ATM it is and I'd like to minimize the amount of code we break.

Reply:
It's also easier for downstream packages if we reduce the number of breaking releases.